### ISSUE TITLE
expose annotation on field value

### ISSUE THREAD

**Issue by Goldziher** [jcrist/msgspec#275]
Hi,

So I am working on integrating Struct inside our internals, but there is a problem - I need access to the raw annotation on struct fields. Currently all I can get is the tree like structure of parsed fields, and if I need an annotation I need to do rather funky manipulation. 

So, can you please expose the raw annotation value on the struct field as part of the field data?

```python
class Field(msgspec.Struct):
    """A record describing a field in an object-like type.

    Parameters
    ----------
    name: str
        The field name as seen by Python code (e.g. ``field_one``).
    encode_name: str
        The name used when encoding/decoding the field. This may differ if
        the field is renamed (e.g. ``fieldOne``).
    type: Type
        The field type.
    required: bool, optional
        Whether the field is required. Note that if `required` is False doesn't
        necessarily mean that `default` or `default_factory` will be set -
        optional fields may exist with no default value.
    default: Any, optional
        A default value for the field. Will be `UNSET` if no default value is set.
    default_factory: Any, optional
        A callable that creates a default value for the field. Will be
        `UNSET` if no ``default_factory`` is set.
    """
    
    annotation: Any # <--- can also be typed with `type` if preferable
    name: str
    encode_name: str
    type: Type
    required: bool = True
    default: Any = UNSET
    default_factory: Any = UNSET
```

**Comment by Goldziher** [comment]
Would also be great to access any metadata available on the field, to reduce the number of calls.

**Comment by jcrist** [comment]
`msgspec.inspect` converts any msgspec-compatible type annotation into a tree of `Type` values, normalizing all the different ways python type annotations may be spelled (e.g. no need to differentiate between `typing.Tuple`, `tuple`, and `tuple[Any, ...]`) into a consistent object structure. It also extracts all the metadata from `Annotated` types and presents it as part of this view. It sounds like that's not what you wanted in #227.

Rather than adding annotations and other things that don't really fit with the design of `msgspec.inspect`, I've pushed up #282 which is a different take on this problem.

- It only supports `Struct` types
- It doesn't recurse into the fields, the raw type annotation for a field is stored on the `StructField` type unedited.
- It doesn't do anything with `msgspec.Meta` annotations. Any consumer will need to properly handle `typing.Annotated` types, and extract `msgspec.Meta` metadata themselves if needed.

Can you take a look at #282 and see if it solves your problem? I don't plan on merging that until I get a confirmation from you that it's what you're looking for.

**Comment by Goldziher** [comment]
Thanks! 

I'll plug that PR in. It will take me a few days probably, but I'll keep you posted.

**Comment by jcrist** [comment]
Fixed by #330. Closing.
